instrumentation/hibernate/hibernate-4.0/javaagent/src/test/java/spring/jpa/SpringJpaTest.java

```diff
@@ -0,0 +1,462 @@
+                        .hasAttributesSatisfyingExactly(
+                            satisfies(
+                                stringKey("hibernate.session_id"),
+                                val -> val.isInstanceOf(String.class))),
+                span ->
+                    span.hasName("SELECT test.Customer")
+                        .hasKind(CLIENT)
+                        .hasParent(trace.getSpan(1))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "hsqldb"),
+                            equalTo(SemanticAttributes.DB_NAME, "test"),
+                            equalTo(SemanticAttributes.DB_USER, "sa"),
+                            equalTo(SemanticAttributes.DB_CONNECTION_STRING, "hsqldb:mem:"),
+                            satisfies(
+                                SemanticAttributes.DB_STATEMENT,
+                                val ->
+                                    val.matches(
+                                        Pattern.compile(
+                                            "select ([^.]+).id([^,]*), ([^.]+).firstName([^,]*), ([^.]+).lastName (.*)from Customer (.*)where ([^.]+).id=\\?"))),
+                            equalTo(SemanticAttributes.DB_OPERATION, "SELECT"),
+                            equalTo(SemanticAttributes.DB_SQL_TABLE, "Customer")),
+                span ->
+                    span.hasName("Transaction.commit")
+                        .hasKind(INTERNAL)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(
+                                stringKey("hibernate.session_id"),
+                                trace
+                                    .getSpan(1)
+                                    .getAttributes()
+                                    .get(stringKey("hibernate.session_id")))),
+                span ->
+                    span.hasName("UPDATE test.Customer")
+                        .hasKind(CLIENT)
+                        .hasParent(trace.getSpan(3))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "hsqldb"),
+                            equalTo(SemanticAttributes.DB_NAME, "test"),
+                            equalTo(SemanticAttributes.DB_USER, "sa"),
+                            equalTo(SemanticAttributes.DB_CONNECTION_STRING, "hsqldb:mem:"),
+                            equalTo(
+                                SemanticAttributes.DB_STATEMENT,
+                                "update Customer set firstName=?, lastName=? where id=?"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "UPDATE"),
+                            equalTo(SemanticAttributes.DB_SQL_TABLE, "Customer"))));
+
+    testing.clearData();
+
+    customer = testing.runWithSpan("parent", () -> repo.findByLastName("Anonymous").get(0));
```

Comment:
Nit: instead of all the `deleteCustomer = customer` local vars, WDYT about adding a local variable for the found customer instead? `customer` would then be effectively final and you would be able to pass it to lambdas, and you'd have just one extra var instead of 3.

Reply:
great suggestion, thank you!